autopaho/extensions/rpc/rpc.go

```diff
@@ -79,13 +79,17 @@ func (h *Handler) Request(ctx context.Context, pb *paho.Publish) (*paho.Publish,
 	pb.Properties.ResponseTopic = h.responseTopic
 	pb.Retain = false
 
-	_, err := h.cm.Publish(ctx, pb)
+	_, err = h.cm.Publish(ctx, pb)
 	if err != nil {
 		return nil, err
 	}
 
-	resp := <-rChan
-	return resp, nil
+	select {
+	case <-ctx.Done():
+		return nil, fmt.Errorf("context ended")
```

Comment:
I think it would be better to return `ctx.Error()` here (its fairly common to check for context errors).